test/e2e/framework/node/resource.go

```diff
@@ -463,12 +463,15 @@ func isNodeUntaintedWithNonblocking(node *v1.Node, nonblockingTaints string) boo
 		nodeInfo.SetNode(node)
 	}
 
-	fit, _, err := predicates.PodToleratesNodeTaints(fakePod, nil, nodeInfo)
+	taints, err := nodeInfo.Taints()
 	if err != nil {
 		e2elog.Failf("Can't test predicates for node %s: %v", node.Name, err)
 		return false
 	}
-	return fit
+
+	return v1helper.TolerationsTolerateTaintsWithFilter(fakePod.Spec.Tolerations, taints, func(t *v1.Taint) bool {
```

Comment:
wishing that long term some of the helpers can move either under `k8s.io/api/core/v1` or somewhere else to help the framework can stop depending on `k8s.io/kubernetes`.

for instance we have `ToleratesTaint()` https://github.com/kubernetes/api/blob/master/core/v1/toleration.go#L37 which is an utility in the api repository, yet the majority of the utilites for `v1.Taint` are under `k8s.io/kubernetes/pkg`

/lgtm
/approve